usr/src/cmd/fm/modules/common/zfs-diagnosis/zfs_de.c

```diff
@@ -796,6 +827,39 @@ zfs_fm_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl, const char *class)
 	if (fmd_case_solved(hdl, zcp->zc_case))
 		return;
 
+	zhdl = fmd_hdl_getspecific(hdl);
+
+	/*
+	 * Find the corresponding pool.
+	 */
+	cb.cb_guid = pool_guid;
+	cb.cb_zhp = NULL;
+	if (zhdl != NULL && zpool_iter(zhdl, find_pool, &cb) != 0)
+		return;
```

Comment:
Should we bump a stat or log a debug message for this error case?

Reply:
I added a stat for this. Does that look good? I think we should only hit this case when the pool iteration itself fails, which should be quite rare.

Reply:
Yep, that looks good.   Thanks.